### DISCUSSION TITLE
nuxi build: (!) Some chunks are larger than 500 KiB after minification.

### DISCUSSION THREAD

**Discussion by anburocky3** [nuxt/nuxt#15935] [Questions]
Everytime, i build it using `yarn build` (`nuxi build`), i see this warning in the console. 

Why does it show it? we are just using `import`  on overall project right?
For source code & reproduction, This is my example [Project repo](https://github.com/anburocky3/iragugal-web)

```
 WARN                                                                                                                                                            15:32:15  
(!) Some chunks are larger than 500 KiB after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/guide/en/#outputmanualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
```

**Comment by anburocky3** [comment]
Any idea, why it happens? @danielroe ?

**Comment by akahon** [comment]
If you just want to get rid of the warning, then you can change the chunk size in the nuxt config file chunkSizeWarningLimit.

I want to reduce (minify) the size of the chunk file, I set minify="terser" but it doesn't work yet :(

```
export default defineNuxtConfig({
  builder: 'vite,
  vite: {
    build: {
      chunkSizeWarningLimit: 1000
    }
  }
})
```

**Reply by TheAlexLichter** [reply]
I'd avise against removing the warning because it only hides the actual problem.

**Comment by TheAlexLichter** [comment]
Can you check if it behaves the same on the latest nuxt3 release (edge, not RC)?

**Reply by anburocky3** [reply]
I'm on `Nuxt 3.0.0-rc.9-27700867.9eb95e2` and still see this error while building.

![image](https://user-images.githubusercontent.com/15264938/188078403-3f634803-c34d-4917-8adc-151e1c8b376c.png)

**Comment by mohkoma** [comment]
Other solution available is to split the node_modules using the manual chunks

```
export default defineConfig({
....
build: {
        rollupOptions: {
            output:{
                manualChunks(id) {
                    if (id.includes('node_modules')) {
                        return id.toString().split('node_modules/')[1].split('/')[0].toString();
                    }
                }
            }
        }
    }
});
```

**Reply by richardkenneth10** [reply]
Do you have the nuxt3 implementation of this?

**Reply by yanghoxom** [reply]
this config create many empty chunk, and bundle size larger than
without this, my bundle is 1.68 MB (204 kB gzip)
with this, my bundle is 2.9 MB (224 kB gzip)

**Comment by oleghalin** [comment]
How can i resolve issue with entry modules? I face next error: 
`Entry module cannot be external `

**Reply by oleghalin** [reply]
Will add more context, i want externalise my @sentry module from entry chunk.

**Reply by TheAlexLichter** [reply]
Better create a new discussion for that.

**Reply by oleghalin** [reply]
nuxt/framework#8110 done

**Comment by praveshishere** [comment]
I think making separate chunks for all firebase modules would work.

> nuxt.config.ts
```ts
// https://v3.nuxtjs.org/docs/directory-structure/nuxt.config

const firebaseModules = ["firebase/app", "firebase/auth", "firebase/firestore"];

export default defineNuxtConfig({
  buildModules: ['@pinia/nuxt'],
  modules: ['@nuxtjs/tailwindcss'],

  vite: {
    build:{
      rollupOptions: {
        output: {
          manualChunks(id) {
            const seprateModule = firebaseModules.find(module => id.includes(module));
            if (seprateModule) return seprateModule;
          }
        }
      }   
    }
  }
});

```

The bundles about which roll-up is warning are firebase modules, so we can just separate them, to resolve the issue.
You could also lazy load firebase in case need an extra performance boost


Now, if we run analyze, firebase is split into 3 parts.
![image](https://user-images.githubusercontent.com/95841565/195862440-170f0e74-ff8d-4fc7-80d6-44d4763d1600.png)

**Comment by YassineBouchama1** [comment]
> Everytime, i build it using `yarn build` (`nuxi build`), i see this warning in the console.
> 
> Why does it show it? we are just using `import` on overall project right? For source code & reproduction, This is my example [Project repo](https://github.com/anburocky3/iragugal-web)
> 
> ```
>  WARN                                                                                                                                                            15:32:15  
> (!) Some chunks are larger than 500 KiB after minification. Consider:
> - Using dynamic import() to code-split the application
> - Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/guide/en/#outputmanualchunks
> - Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
> ```


go to vite.config.js
and put this 

`  build: {
    chunkSizeWarningLimit: 1600,
  },`


Full Code
`import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'

export default defineConfig({
  plugins: [react()],
  build: {
    chunkSizeWarningLimit: 1600,
  },
})`

**Reply by Ashu-r** [reply]
This just suppresses the warning. Please refer to https://github.com/nuxt/nuxt/discussions/15935#discussioncomment-3308074

**Reply by joaopedrodcf** [reply]
I was having this exact issue and the only was of solving it was with this reply from a reply of @danielroe in another thread

https://github.com/nuxt/nuxt/issues/22127#issuecomment-1635925362

```ts
export default defineNuxtConfig({
  hooks: {
    "vite:extendConfig"(config, { isClient })  {
      if (isClient) {
        config.build.rollupOptions.output.manualChunks = function (_id) {
          return 'index';
        }
      }
    }
  }
});
```

**Reply by joaopedrodcf** [reply]
You then can do any sort of optimzation on the manualChunks a vendor with everything, or everything separate, etc
